src/common/vscode-settings/index.ts

```diff
@@ -1,2 +1,48 @@
-export {VsCodeSettings} from './vscode-settings'
-export {WithVSCode} from './with-vscode'
+import type {NodeProject} from 'projen/lib/javascript'
+
+/**
+ * Adds VsCode settings and recommended extensions to the project, unless vscode is disabled.
+ *
+ * NOTE: It can either explicitly disabled with the option `vscode: false`,
+ * or automatically for projects that have a parent, so that only the root project has the settings.
+ */
+export const addVsCodeToProject = ({vscode}: NodeProject) => {
```

Comment:
@quesabe nit pick — I would keep `project` as an input, looks more verbose to me, but makes things more clear:
```suggestion
export const addVsCodeToProject = (project: NodeProject) => {
```

Reply:
I've also removed comment inside the functions, since the function body is kinda short and self-explanatory.